plugins/techdocs-backend/src/service/router.ts

```diff
@@ -149,53 +154,35 @@ export async function createRouter({
           // add the publisher type in the list here.
-          if (!(await publisher.hasDocsBeenGenerated(entity))) {
-            logger.info(
-              'No pre-generated documentation files found for the entity in the storage. Building docs...',
-            );
-            await docsBuilder.build();
-            // With a maximum of ~5 seconds wait, check if the files got published and if docs will be fetched
-            // on the user's page. If not, respond with a message asking them to check back later.
-            // The delay here is to make sure GCS/AWS/etc. registers newly uploaded files which is usually <1 second
-            let foundDocs = false;
-            for (let attempt = 0; attempt < 5; attempt++) {
-              if (await publisher.hasDocsBeenGenerated(entity)) {
-                foundDocs = true;
-                break;
-              }
-              await new Promise(r => setTimeout(r, 1000));
-            }
-            if (!foundDocs) {
-              logger.error(
-                'Published files are taking longer to show up in storage. Something went wrong.',
-              );
-              res
-                .status(408)
-                .send(
-                  'Sorry! It is taking longer for the generated docs to show up in storage. Check back later.',
-                );
-              return;
+          await docsBuilder.build();
+          // With a maximum of ~5 seconds wait, check if the files got published and if docs will be fetched
+          // on the user's page. If not, respond with a message asking them to check back later.
+          // The delay here is to make sure GCS/AWS/etc. registers newly uploaded files which is usually <1 second
+          for (let attempt = 0; attempt < 5; attempt++) {
+            if (await publisher.hasDocsBeenGenerated(entity)) {
+              foundDocs = true;
+              break;
             }
-          } else {
-            logger.info(
-              'Found pre-generated docs for this entity. Serving them.',
+            await new Promise(r => setTimeout(r, 1000));
+          }
+          if (!foundDocs) {
+            logger.error(
+              'Published files are taking longer to show up in storage. Something went wrong.',
             );
-            // TODO: re-trigger build for cache invalidation.
-            // Add build info in techdocs_metadata.json and compare it against
-            // the etag/commit in the repository.
-            // Without this, docs will not be re-built once they have been generated.
-            // Although it is unconventional that anyone will face this issue - because
-            // if you have an external storage, you should be using CI/CD to build and publish docs.
+            res.status(408).json({
+              error:
+                'Sorry! It is taking longer for the generated docs to show up in storage. Check back later.',
```

Comment:
to an end user "taking longer" infers it's longer than some period I would be familiar with, which may not be valid in this case to users viewing documentation? (what if this is the first doc they've ever seen, etc)  I think this might need to be more generic, and just say something like,

```suggestion
                'Sorry! It took too long for the generated docs to show up in storage. Check back later.',
```